pkg/apis/pipeline/v1beta1/resolver_types.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+// ResolverRef can be used to refer to a Pipeline or Task in a remote
+// location like a git repo. This feature is in alpha and these fields
+// are only available when the alpha feature gate is enabled.
+type ResolverRef struct {
+	// Resolver is the name of the resolver that should perform
+	// resolution of the referenced Tekton resource, such as "git".
+	// +optional
+	Resolver string `json:"resolver,omitempty"`
+	// Resource contains the parameters used to identify the
+	// referenced Tekton resource. Example entries might include
+	// "repo" or "path" but the set of params ultimately depends on
+	// the chosen resolver.
+	// +optional
+	Resource map[string]string `json:"resource,omitempty"`
```

Comment:
I'm wondering about using`map[string]string` for `Resource` - could we use `Parameters` instead?
Especially considering that the proposal suggests using object `Parameters` in this field when it's available - maybe using a list of string `Parameters` here would help in the transition later? 

> Note that the fields under `resource:` are validated by a resolver, not by Tekton Pipelines. [TEP-0075 (Dictionary
Params)](https://github.com/tektoncd/community/pull/479) describes adding support for JSON object schema to Params. It would make sense to bring in the same schema support for the `resource:` field as well and eventually resolvers might be able to publish the schema they accept so that Pipelines can enforce it during validation of a `TaskRun` or `PipelineRun`. 
~ [TEP-0060](https://github.com/tektoncd/community/blob/main/teps/0060-remote-resource-resolution.md#1-establish-a-new-syntax-in-tekton-pipelines-taskref-and-pipelineref-structs-for-remote-resources)

K8s API convention encouraging use of sub-objects instead of maps: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps

Reply:
Thanks for the feedback. I've changed this to use the following syntax:

```yaml
taskRef:
  remote: git
  resource:
  - name: repo
    value: https://github.com/tektoncd/catalog.git
  - name: path
    value: /task/golang-build/0.3/golang-build.yaml
```